### PR TITLE
fix: fix group by on chunked segment

### PR DESCRIPTION
**Pull request by sunby**
https://github.com/milvus-io/milvus/issues/37244

### CODE REVIEW

File: internal/core/src/exec/operator/groupby/SearchGroupByOperator.h [hunk 1]
```diff
@@ -140,7 +140,7 @@ PrepareVectorIteratorsFromIndex(const SearchInfo& search_info,
                     index.VectorIterators(dataset, search_conf, bitset);
             if (iterators_val.has_value()) {
                 search_result.AssembleChunkVectorIterators(
-                    nq, 1, -1, iterators_val.value());
+                    nq, 1, {0}, iterators_val.value());
```
Comment (PwzXxm) [inline internal/core/src/exec/operator/groupby/SearchGroupByOperator.h:143]: `{}`?

Comment (PwzXxm) [inline internal/core/src/exec/operator/groupby/SearchGroupByOperator.h:143]: This change results in different behavior. Change this to `{}` or check `if (total_rows_until_chunk_.size() == 1` in the function `convert_to_segment_offset()`

File: internal/core/src/query/SearchOnSealed.cpp [hunk 2]
```diff
@@ -201,7 +203,7 @@ SearchOnSealed(const Schema& schema,
         auto sub_qr = BruteForceSearchIterators(
             dataset, vec_data, row_count, search_info, bitset, data_type);
         result.AssembleChunkVectorIterators(
-            num_queries, 1, -1, sub_qr.chunk_iterators());
+            num_queries, 1, {0}, sub_qr.chunk_iterators());
```
Comment (PwzXxm) [inline internal/core/src/query/SearchOnSealed.cpp:206]: ditto

File: internal/core/src/exec/operator/groupby/SearchGroupByOperator.h (hunk 1 above)
Comment (sunby) [reply internal/core/src/exec/operator/groupby/SearchGroupByOperator.h:143]: > This change results in different behavior. Change this to `{}` or check `if (total_rows_until_chunk_.size() == 1` in the function `convert_to_segment_offset()`

ok, this does not affect correctness and I will change it in next pr.

**Comment by PwzXxm** [comment]
Discussed. Fixed in next PR.

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**